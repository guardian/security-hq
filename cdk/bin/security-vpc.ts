import 'source-map-support/register';
import { App } from 'aws-cdk-lib';
import { SecurityVpc } from '../lib/security-vpc';

// Note, this stack is synthed locally rather than in CI as it is
// 'environment-aware'
// (https://docs.aws.amazon.com/cdk/latest/guide/environments.html) and so
// requires AWS crendentials. Ideally, longer-term we'd support this in Github
// Actions somehow.
const app = new App();
new SecurityVpc(app, 'security-vpc', {
  stack: 'security',
  stage: 'PROD',
  env: { region: 'eu-west-1', account: process.env.CDK_DEFAULT_ACCOUNT }, // Crucial to ensure VPC uses all AZs.
});
