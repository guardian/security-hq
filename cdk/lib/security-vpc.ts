import type { App } from '@aws-cdk/core';
import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack } from '@guardian/cdk/lib/constructs/core';
import { GuKinesisStream } from '@guardian/cdk/lib/constructs/kinesis';

export class SecurityVpc extends GuStack {
  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);

    new GuKinesisStream(this, 'Never gonna happen', {});
    // new GuVpc(this, 'SecurityVpc', {}); // TODO specify CIDR
  }
}
