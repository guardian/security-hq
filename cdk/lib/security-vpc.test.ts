import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import { App } from '@aws-cdk/core';
import { SecurityVpc } from './security-vpc';

describe('VPC stack', () => {
  it('matches the snapshot', () => {
    const app = new App();
    const stack = new SecurityVpc(app, 'security-vpc', { stack: 'security' });
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});
