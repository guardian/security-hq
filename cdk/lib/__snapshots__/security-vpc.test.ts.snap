// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VPC stack matches the snapshot 1`] = `
Object {
  "Conditions": Object {
    "AwsCdkKinesisEncryptedStreamsUnsupportedRegions": Object {
      "Fn::Or": Array [
        Object {
          "Fn::Equals": Array [
            Object {
              "Ref": "AWS::Region",
            },
            "cn-north-1",
          ],
        },
        Object {
          "Fn::Equals": Array [
            Object {
              "Ref": "AWS::Region",
            },
            "cn-northwest-1",
          ],
        },
      ],
    },
  },
  "Parameters": Object {
    "Stage": Object {
      "AllowedValues": Array [
        "CODE",
        "PROD",
      ],
      "Default": "CODE",
      "Description": "Stage name",
      "Type": "String",
    },
  },
  "Resources": Object {
    "NevergonnahappenF0CA0A60": Object {
      "Properties": Object {
        "RetentionPeriodHours": 24,
        "ShardCount": 1,
        "StreamEncryption": Object {
          "Fn::If": Array [
            "AwsCdkKinesisEncryptedStreamsUnsupportedRegions",
            Object {
              "Ref": "AWS::NoValue",
            },
            Object {
              "EncryptionType": "KMS",
              "KeyId": "alias/aws/kinesis",
            },
          ],
        },
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/security-hq",
          },
          Object {
            "Key": "Stack",
            "Value": "security",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
      },
      "Type": "AWS::Kinesis::Stream",
    },
  },
}
`;
